src/main/java/org/prebid/server/settings/S3ApplicationSettings.java

```diff
@@ -0,0 +1,205 @@
+                        fileContentOpt.map(fileContent -> jacksonMapper.decodeValue(fileContent, Account.class)))
+                .compose(accountOpt -> {
+                    if (accountOpt.isPresent()) {
+                        return Future.succeededFuture(accountOpt.get());
+                    } else {
+                        return Future
+                                .failedFuture(new PreBidException("Account with id %s not found".formatted(accountId)));
+                    }
+                })
+                .recover(ex -> {
+                    if (ex instanceof DecodeException) {
+                        return Future
+                                .failedFuture(
+                                        new PreBidException(
+                                                "Invalid json for account with id %s".formatted(accountId)));
+                    }
+                    return Future
+                            .failedFuture(new PreBidException("Account with id %s not found".formatted(accountId)));
+                });
+    }
+
+    @Override
+    public Future<StoredDataResult> getStoredData(
+            String accountId,
+            Set<String> requestIds,
+            Set<String> impIds,
+            Timeout timeout) {
+
+        return getFileContents(storedRequestsDirectory, requestIds).compose(storedIdToRequest ->
+                getFileContents(storedImpressionsDirectory, impIds)
+                        .map(storedIdToImp -> {
+                                    final List<String> missingStoredRequestIds =
+                                            getMissingStoredDataIds(storedIdToRequest).stream()
+                                                    .map("No stored request found for id: %s"::formatted).toList();
+                                    final List<String> missingStoredImpressionIds =
+                                            getMissingStoredDataIds(storedIdToImp).stream()
+                                                    .map("No stored impression found for id: %s"::formatted).toList();
+
+                                    return StoredDataResult.of(
+                                            filterOptionalFileContent(storedIdToRequest),
+                                            filterOptionalFileContent(storedIdToImp),
+                                            Stream.concat(
+                                                    missingStoredImpressionIds.stream(),
+                                                    missingStoredRequestIds.stream()).toList());
+                                }
+                        ));
+    }
+
+    private Map<String, String> filterOptionalFileContent(Map<String, Optional<String>> fileContents) {
+        return fileContents
+                .entrySet()
+                .stream()
+                .filter(e -> e.getValue().isPresent())
+                .collect(Collectors.toMap(Map.Entry::getKey, e -> e.getValue().get()));
+    }
```

Comment:
Please, don't store optionals inside map, because there is no distinction between absence and `null` value, since both are bad for us. Just use simple `Map<String, String>`. 
and then you can simplify this to:
```java
private Map<String, String> filterEmptyContent(Map<String, Optional<String>> fileContents) {
        return fileContents.entrySet().stream()
                .filter(Objects::nonNull)
                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue);
    }
```

Also, please, note how I placed calls to `.entrySet().stream()` on the same line where I used `fileContents`. This is just our preferred styling.

Reply:
I refactored this a little bit to get rid of the `filterEmptyContent` method

Reply:
I'm also curious about your formatting. How do you automatically format your files?
Checkstyle in intellij is not very useful as it does not format, but only highlights issues (if it's in the mode, because some rules it doens't understand ).

Manually formatting stuff seems so 2000 :laughing:

Reply:
Sadly, yes, we format code by hand :(. As you may have noticed, codebase uses not like "default" Java conventions, so we can't just plug in some default formatter.